diff --git a/configure/FindMySQL.cmake b/configure/FindMySQL.cmake
index 7896ed7..47606db 100644
--- a/configure/FindMySQL.cmake
+++ b/configure/FindMySQL.cmake
@@ -38,29 +38,6 @@ else()
             "/usr/mysql/lib64/mysql" )
 endif()
 
-if(NOT DEFINED DB_CLIENT_COMPILE)
-  try_run(DB_CLIENT_COMPILE
-          DB_CLIENT_RUN
-          ${CMAKE_CURRENT_BINARY_DIR}
-          ${CMAKE_CURRENT_SOURCE_DIR}/configure/test_db_client.cpp
-          COMPILE_DEFINITIONS "-I ${MYSQL_INCLUDE_DIR}"
-          LINK_LIBRARIES "${MYSQL_LIBRARY}"
-          COMPILE_OUTPUT_VARIABLE DB_CLIENT_COMPILE_OUTPUT
-          RUN_OUTPUT_VARIABLE DB_CLIENT_VERSION)
-
-  set(msg "Check if we can compile a simple database client program: ${DB_CLIENT_COMPILE}")
-
-  if(NOT DB_CLIENT_COMPILE)
-    message(STATUS ${msg})
-  else()
-    message(FATAL_ERROR ${msg} "\n\nBuild output:\n" ${DB_CLIENT_COMPILE_OUTPUT})
-  endif()
-
-  string(STRIP ${DB_CLIENT_VERSION} DB_CLIENT_VERSION)
-  set(TDB_DB_CLIENT_VERSION ${DB_CLIENT_VERSION} CACHE INTERNAL "database client version")
-
-  message("Output: ${TDB_DB_CLIENT_VERSION}")
-endif()
 
 # handle the QUIETLY and REQUIRED arguments and set MYSQL_FOUND to TRUE if
 # all listed variables are TRUE
